src/main/java/com/worksoft/jenkinsci/plugins/ctm/model/CTMServer.java

```diff
@@ -13,528 +13,613 @@
-            .header("jsonOrXml", "json");
-
-    httpRequest.body("");
-
-    CTMResult result = sendRequest(httpRequest);
-
-    return result;
-  }
-
- */
-
-  private CTMResult sendRequest (HttpRequest request) throws HttpException {
-
-    CTMResult result;
-
-
-    try {
-      if (auth != null) {
-        request.tokenAuthentication(auth.getAccess_token());
-      }
-
-      HttpResponse response = request.send();
-      result = lastCTMResult = new CTMResult(response);
-      if (result.is200()) {
-//      status = true;
-
-      } else if (result.statusCode() == 401) {
-//      status = false;
-        throw new Exception("Unauthorized");
-//      payload = Unauthorized;
-      } else {
-
-        log.warning("CTM/Portal/api failed " + response.toString(true));
-//      status = false;
-//      if (payload != null) {
-//        payload = response.statusPhrase();
-//      }
-      }
-
-    } catch (Throwable t) {
-      result = new CTMResult(null);
-      log.severe("ERROR: unexpected error while processing request: " + request);
-      log.severe("ERROR: exception: " + t);
-      log.severe("ERROR: exception: " + t.getMessage());
-      log.severe("ERROR: stack trace:  ");
-      StackTraceUtils.printSanitizedStackTrace(t.getCause());
-      throw new HttpException(t);
-    }
-
-    return result;
-  }
-}
+ package com.worksoft.jenkinsci.plugins.ctm.model;
+
+ import com.cloudbees.plugins.credentials.common.UsernamePasswordCredentials;
+ import com.worksoft.jenkinsci.plugins.ctm.CTMExecute.JobDetails;
+ 
+ import jodd.http.HttpException;
+ import jodd.http.HttpRequest;
+ import jodd.http.HttpResponse;
+ import net.sf.json.JSONArray;
+ import net.sf.json.JSONObject;
+ import org.apache.commons.lang.StringUtils;
+ import org.codehaus.groovy.runtime.StackTraceUtils;
+ 
+ import java.io.UnsupportedEncodingException;
+ //import java.net.URL;
+ import java.text.DateFormat;
+ import java.text.SimpleDateFormat;
+ import java.util.*;
+ import java.util.logging.Logger;
+ import java.util.stream.Collectors;
+ 
+ public class CTMServer {
+ 
+   private static final Logger log = Logger.getLogger("jenkins.wsCTMServer");
+ 
+   private final String portalUrl;
+   private final UsernamePasswordCredentials credentials;
+ 
+   private Auth auth;
+ 
+   private String authenticationUrl = "";
+   private String ctmUrl = "";
+   private CTMResult lastCTMResult;
+   // private static final String MMddyyyy = "MM/dd/yyyy";
+   private static final String yyyyMMdd = "yyyy-MM-dd";
+   private static final int PAGE_SIZE = 100;
+ 
+   public CTMResult getLastCTMResult() {
+     return lastCTMResult;
+   }
+ 
+   public CTMServer(String url, UsernamePasswordCredentials credentials) {
+     if (credentials == null) {
+       throw new RuntimeException("Credentials must be provided!");
+     }
+ 
+     try {
+       // URL foo = new URL(url);
+     } catch (Exception e) {
+       throw new RuntimeException("Execution Manager URL is invalid " + e.getMessage());
```

Comment:
This probably should say Continuous Testing Manager instead of Execution Manager.    Also, we probably want to keep the MalformedURLException and keep this message in the exception that we throw.     Then you can add a fallback catch(Exception) - with a different message.    Always saying that it is a URL invalid for a global exception will lead to confusion.

Reply:
Text update from "Execution Manager" to "Continuous Testing Manager" done